### DISCUSSION TITLE
Is there a way to export a function that returns a Result?

### DISCUSSION THREAD

**Discussion by ryb73** [wasm-bindgen/wasm-bindgen#3950] [Q&A]
A wasm_bindgen function that in Rust returns a Result will generate a function that in JS unwraps the result, returning the Ok value and throwing the Err value. For example:
```rust
#[wasm_bindgen(js_name = extractReferences)]
pub fn extract_references_wasm(sql: &str) -> Result<ExtractedReferences, String> {
    extract_references(sql).into()
}
```
Exported definition in .d.ts:
```typescript
export function extractReferences(sql: string): ExtractedReferences;
```

What I want is not for the error to be thrown, but to simply return a Result object:
```typescript
export function extractReferences(sql: string): Result<ExtractedReferences, string>;
```

Is there a way to do this with wasm_bindgen?

**Comment by daxpedda** [comment]
See https://github.com/rustwasm/wasm-bindgen/issues/2407, which would probably be needed for something like this.
